Author: Sergey Slipchenko <faergeek@gmail.com>
Description: Remove missing files to pass check target

diff -Naurp lxpanel.orig/po/POTFILES.in lxpanel/po/POTFILES.in
--- lxpanel.orig/po/POTFILES.in	2012-04-20 17:12:48.420543687 +0200
+++ lxpanel/po/POTFILES.in	2012-04-20 17:14:32.617254186 +0200
@@ -17,7 +17,6 @@ data/ui/panel-pref.ui
 # Putting translation of plugins in main catalog is not a good idea...
 src/plugins/cpu/cpu.c
 src/plugins/deskno/deskno.c
-src/plugins/image.c
 src/plugins/launchbar.c
 src/plugins/taskbar.c
 src/plugins/dclock.c
@@ -35,10 +34,6 @@ src/plugins/volume/volume.c
 src/plugins/volume/volume-impl.c
 src/plugins/volumealsa/volumealsa.c
 
-src/plugins/netstat/netstat.c
-src/plugins/netstat/passwd_gui.c
-src/plugins/netstat/lxnm_client.c
-
 src/plugins/netstatus/netstatus.c
 src/plugins/netstatus/netstatus-dialog.c
 src/plugins/netstatus/netstatus-icon.c
