Author: Sergey Slipchenko <faergeek@gmail.com>
Description: Remove missing files to pass check target
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -14,7 +14,6 @@ data/ui/panel-pref.ui
 # Putting translation of plugins in main catalog is not a good idea...
 src/plugins/cpu/cpu.c
 src/plugins/deskno/deskno.c
-src/plugins/image.c
 src/plugins/launchbar.c
 src/plugins/taskbar.c
 src/plugins/dclock.c
@@ -32,17 +31,12 @@ src/plugins/volume/volume.c
 src/plugins/volume/volume-impl.c
 src/plugins/volumealsa/volumealsa.c
 
-src/plugins/netstat/netstat.c
-src/plugins/netstat/passwd_gui.c
-src/plugins/netstat/lxnm_client.c
-
 src/plugins/netstatus/netstatus.c
 src/plugins/netstatus/netstatus-dialog.c
 src/plugins/netstatus/netstatus-icon.c
 src/plugins/netstatus/netstatus-iface.c
 src/plugins/netstatus/netstatus-sysdeps.c
 src/plugins/netstatus/netstatus-util.c
-src/plugins/netstatus/gnome-netstatus.glade
 
 src/plugins/batt/batt.c
 
